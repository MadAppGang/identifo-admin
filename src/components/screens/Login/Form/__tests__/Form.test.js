import { LoginForm } from '..';

describe('<LoginForm />', () => {
  let props;

  beforeEach(() => {
    props = {
      login: jest.fn(),
      resetError: jest.fn(),
      signingIn: false,
      error: null,
    };
  });

  test('renders as expected', () => {
    expect(shallow(<LoginForm {...props} />)).toMatchSnapshot();
  });

  test('renders as expected when signing in', () => {
    props.signingIn = true;
    expect(shallow(<LoginForm {...props} />)).toMatchSnapshot();
  });

  test('renders as expected when error', () => {
    props.error = new Error('message');
    expect(shallow(<LoginForm {...props} />)).toMatchSnapshot();
  });

  test('calls login on form submit', () => {
    const component = shallow(<LoginForm {...props} />);

    component.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(props.login).toHaveBeenCalled();
  });

  test('calls login with credentials from inputs on submit button click', () => {
    const changeEvent = (name, value) => ({ target: { name, value } });
    const component = shallow(<LoginForm {...props} />);

    component
      .find('Input[name="email"]')
      .simulate('change', changeEvent('email', 'Email'));

    component
      .find('Input[name="password"]')
      .simulate('change', changeEvent('password', 'Password'));

    component.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(props.login).toHaveBeenCalledWith('Email', 'Password');
  });

  test('calls resetError on error message click', () => {
    props.error = new Error('message');
    const component = shallow(<LoginForm {...props} />);

    component.find('LoginErrorMessage').simulate('click');
    expect(props.resetError).toHaveBeenCalled();
  });
});
