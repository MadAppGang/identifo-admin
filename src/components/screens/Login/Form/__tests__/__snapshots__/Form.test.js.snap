// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders as expected 1`] = `
<form
  className="iap-login-form"
  onSubmit={[Function]}
>
  <p
    className="iap-login-form__title"
  >
    Identifo
  </p>
  <Input
    Icon={null}
    errorMessage=""
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />
  <Input
    Icon={null}
    errorMessage=""
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <LoginButton
    loading={false}
    onClick={null}
  />
</form>
`;

exports[`<LoginForm /> renders as expected when error 1`] = `
<form
  className="iap-login-form"
  onSubmit={[Function]}
>
  <p
    className="iap-login-form__title"
  >
    Identifo
  </p>
  <LoginErrorMessage
    message="message"
    onClick={[Function]}
  />
  <Input
    Icon={null}
    errorMessage=""
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />
  <Input
    Icon={null}
    errorMessage=""
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <LoginButton
    loading={false}
    onClick={null}
  />
</form>
`;

exports[`<LoginForm /> renders as expected when signing in 1`] = `
<form
  className="iap-login-form"
  onSubmit={[Function]}
>
  <p
    className="iap-login-form__title"
  >
    Identifo
  </p>
  <Input
    Icon={null}
    errorMessage=""
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />
  <Input
    Icon={null}
    errorMessage=""
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <LoginButton
    loading={true}
    onClick={null}
  />
</form>
`;
